Split OSD tests by CI group

### Is your feature request related to a problem? Please describe

Currently OSD tests are run all together. OSD has a lot of tests that when all run together fail consistently.

### Describe the solution you'd like

We should split the OSD tests by CI groups similar to the OSD CI and now the FTR CI as well

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

When trying to use `CYPRESS_NO_COMMEND_LOG=1` in opensearch-project/opensearch-dashboards-functional-test#1428, we see issues related to this test. It also shows up errors in another approach of opensearch-project/opensearch-build#4796:

```

  Running:  core-opensearch-dashboards/opensearch-dashboards/apps/vis_builder/ba          (26 of 57)
            sic.spec.js


  Visualization Builder Base Tests
    ✓ Show existing visualizations in Visualize and navigate to it (21988ms)
    ✓ Navigate to Visualization Builder from Create Visualization (8311ms)
    1) Create new basic metric visualization
    2) Be able to add/ edit and remove a field
    3) Be able to save a visualization


  2 passing (5m)
  3 failing

  1) Visualization Builder Base Tests
       Create new basic metric visualization:
     AssertionError: Timed out retrying after 60000ms: Expected to find element: `[data-test-subj="homeIcon"]`, but never found it.
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/vis_builder/basic.spec.js:600:13)

  2) Visualization Builder Base Tests
       Be able to add/ edit and remove a field:
     AssertionError: Timed out retrying after 60000ms: Expected to find element: `[data-test-subj="homeIcon"]`, but never found it.
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/vis_builder/basic.spec.js:600:13)

  3) Visualization Builder Base Tests
       Be able to save a visualization:
     AssertionError: Timed out retrying after 60000ms: Expected to find element: `[data-test-subj="homeIcon"]`, but never found it.
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/vis_builder/basic.spec.js:600:13)
```

Related to: https://github.com/opensearch-project/opensearch-dashboards-functional-test/blame/2.15/cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/vis_builder/basic.spec.js#L55-L57

It was suggested by team to reduce the defaultCommandTimeout from 60s to 10s in this PR: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1418/files#diff-db1f8bb89a89d5b089b39a2af3c27ab75092f1844a158eca0a76445d513d9044

Also this change has failed many other OSD plugins tests as documented in https://github.com/opensearch-project/OpenSearch/issues/14442, therefore we revert the above change in https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1421.

There are 2 approaches right now to resolve this:

1. Team push for root cause the exact issue and fix on test cases
2. Infra try to only apply the `defaultCommandTimeout` with 10s just for OSD Core and try to pass the checks.

Thanks.

After more tries I realize my initial implementation of changing 60s to 10s is noop due to cypress only accept int instead of str as value type for `defaultCommandTimeout`.

This means if we apply the 10s correctly, the test will immediately fail from the start due to not enough wait:
```

  Running:  core-opensearch-dashboards/opensearch-dashboards/apps/vis_type_table            (1 of 5)
            /basic.spec.js


  table visualization basic functions
    1) "before all" hook for "Should apply changed params and allow to reset"


  0 passing (26s)
  1 failing

  1) table visualization basic functions
       "before all" hook for "Should apply changed params and allow to reset":
     AssertionError: Timed out retrying after 10000ms: Expected to find element: `[data-test-subj="superDatePickerstartDatePopoverButton"],[data-test-subj="superDatePickerShowDatesButton"]`, but never found it.

Because this error occurred during a `before all` hook we are skipping the remaining tests in the current suite: `table visualization basic f...`
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress/support/index.js:1304:110)
```

So we back to use 60s for all test.

More Windows issues after applying the .gitattributes to fix json formatting:
```
  Running:  core-opensearch-dashboards/opensearch-dashboards/apps/data_explorer/            (5 of 8)
            shared_links.spec.js


  shared links
    shared links with state in query
      1) should allow for copying the snapshot URL
      √ should allow for copying the snapshot URL as a short URL
      2) should allow for copying the saved object URL
    shared links with state in sessionStorage
      √ should allow for copying the snapshot URL (17659ms)
      √ should allow for copying the snapshot URL as a short URL
      √ should allow for copying the saved object URL (13177ms)


  4 passing (3m)
  2 failing

  1) shared links
       shared links with state in query
         should allow for copying the snapshot URL:
     AssertionError: Timed out retrying after 60000ms: expected 'http://localhost:5601/app/data-explorer/discover?security_tenant=global#/?_a=(discover:(columns:!(_source),isDirty:!
f,sort:!()),metadata:(indexPattern:\'logstash-*\',view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:\'2015-09-19T06:31:44.000Z\',to:\'2015-09-23T18:31:44
.000Z\'))&_q=(filters:!(),query:(language:kuery,query:\'\'))' to satisfy [Function]
      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress\integration\core-opensearch-dashboards\opensearch-dashboards\apps\data_explorer\shared_links.spec.js:262:8)

  2) shared links
       shared links with state in query
         should allow for copying the saved object URL:

      Timed out retrying after 60000ms
      + expected - actual

      -'http://localhost:5601/app/data-explorer/discover/#/view/ab12e3c0-f231-11e6-9486-733b1ac9221a?_g=(filters%3A!()%2CrefreshInterval%3A(pause%3A!t%2Cvalue%3A0)%2Ctime%3A(from%3A
\'2015-09-19T06%3A31%3A44.000Z\'%2Cto%3A\'2015-09-23T18%3A31%3A44.000Z\'))'
      +'http://localhost:5601/app/data-explorer/discover/#/view/ab12e3c0-f231-11e6-9486-733b1ac9221a?_g=(filters%3A!()%2CrefreshInterval%3A(pause%3A!t%2Cvalue%3A0)%2Ctime%3A(from%3A
\'2015-09-19T13%3A31%3A44.000Z\'%2Cto%3A\'2015-09-24T01%3A31%3A44.000Z\'))'

      at Context.eval (http://localhost:5601/__cypress/tests?p=cypress\integration\core-opensearch-dashboards\opensearch-dashboards\apps\data_explorer\shared_links.spec.js:281:84)
```

--- a/manifests/2.15.0/opensearch-dashboards-2.15.0-test.yml
+++ b/manifests/2.15.0/opensearch-dashboards-2.15.0-test.yml
@@ -18,6 +18,7 @@ components:
         logging.json: false
         data.search.aggs.shardDelay.enabled: true
         csp.warnLegacyBrowsers: false
+      ci-groups: 9
   - name: alertingDashboards
     integ-test:
       test-configs:

--- a/src/manifests/test_manifest.py
+++ b/src/manifests/test_manifest.py
@@ -77,6 +77,7 @@ class TestManifest(ComponentManifest['TestManifest', 'TestComponents']):
                             },
                             "test-configs": {"type": "list", "allowed": ["with-security", "without-security"]},
                             "additional-cluster-configs": {"type": "dict"},
+                            "ci-groups": {"type": "integer"}
                         },
                     },
                     "bwc-test": {

--- a/src/test_workflow/integ_test/integ_test_runner.py
+++ b/src/test_workflow/integ_test/integ_test_runner.py
@@ -43,10 +43,19 @@ class IntegTestRunner(abc.ABC):
                 if component.name in self.test_manifest.components:
                     test_config = self.test_manifest.components[component.name]
                     if test_config.integ_test:
-                        test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
-                        test_results = test_suite.execute_tests()
-                        [self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
-                        all_results.append(component.name, test_results)
+                        if 'ci-groups' in test_config.integ_test.keys():
+                            orig_component_name = component.name
+                            for i in range(1, test_config.integ_test['ci-groups'] + 1):
+                                component.name = f"{orig_component_name}-ci-group-{i}"
+                                test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
+                                test_results = test_suite.execute_tests()
+                                [self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
+                                all_results.append(component.name, test_results)
+                        else:
+                            test_suite = self.__create_test_suite__(component, test_config, work_dir.path)
+                            test_results = test_suite.execute_tests()
+                            [self.test_recorder.test_results_logs.generate_component_yml(result_data) for result_data in test_suite.result_data]
+                            all_results.append(component.name, test_results)
                     else:
                         logging.info(f"Skipping integ-tests for {component.name}, as it is currently not supported")
                 else:

--- a/tests/tests_test_workflow/test_integ_workflow/integ_test/test_integ_test_runner_opensearch_dashboards.py
+++ b/tests/tests_test_workflow/test_integ_workflow/integ_test/test_integ_test_runner_opensearch_dashboards.py
@@ -6,7 +6,7 @@
 # compatible open source license.
 
 import unittest
-from unittest.mock import MagicMock, Mock, patch
+from unittest.mock import MagicMock, Mock, call, patch
 
 from test_workflow.integ_test.integ_test_runner_opensearch_dashboards import IntegTestRunnerOpenSearchDashboards
 
@@ -90,3 +90,81 @@ class TestIntegTestRunnerOpenSearchDashboards(unittest.TestCase):
             mock_path,
             mock_test_recorder_object
         )
+
+    @patch("test_workflow.integ_test.integ_test_runner_opensearch_dashboards.IntegTestStartPropertiesOpenSearch")
+    @patch("test_workflow.integ_test.integ_test_runner_opensearch_dashboards.IntegTestStartPropertiesOpenSearchDashboards")
+    @patch("test_workflow.integ_test.integ_test_runner_opensearch_dashboards.IntegTestSuiteOpenSearchDashboards")
+    @patch("test_workflow.integ_test.integ_test_runner.TestRecorder")
+    @patch("test_workflow.integ_test.integ_test_runner.TemporaryDirectory")
+    def test_with_integ_test_ci_groups(self, mock_temp: Mock, mock_test_recorder: Mock, mock_suite: Mock, mock_properties: Mock, mock_properties_dependency: Mock) -> None:
+        self.args.paths = {"opensearch-dashboards": "test-path"}
+        self.args.component = "sql"
+        self.args.test_run_id = "12345"
+
+        mock_test_config = MagicMock()
+        mock_test_config.integ_test = {'test-configs': ['with-security'], 'ci-groups': 3}
+        self.test_manifest.components = {"sql": mock_test_config}
+
+        mock_build_manifest = MagicMock()
+        mock_components = MagicMock()
+        mock_component = MagicMock()
+        mock_component.name = "sql"
+        mock_components.select.return_value = [mock_component]
+        mock_build_manifest.components = mock_components
+
+        mock_bundle_manifest = MagicMock()
+        mock_dependency_installer = MagicMock()
+
+        mock_properties_object = MagicMock()
+        mock_properties_object.bundle_manifest = mock_bundle_manifest
+        mock_properties_object.build_manifest = mock_build_manifest
+        mock_properties_object.dependency_installer = mock_dependency_installer
+
+        mock_properties.return_value = mock_properties_object
+
+        mock_properties_dependency_object = MagicMock()
+        mock_properties_dependency_object.bundle_manifest = MagicMock()
+        mock_properties_dependency_object.build_manifest = MagicMock()
+        mock_properties_dependency_object.dependency_installer = MagicMock()
+        mock_properties_dependency.return_value = mock_properties_dependency_object
+
+        mock_suite_object = MagicMock()
+        mock_test_results = MagicMock()
+
+        mock_suite_object.execute_tests.return_value = mock_test_results
+
+        mock_suite.return_value = mock_suite_object
+
+        mock_path = MagicMock()
+        mock_temp.return_value.__enter__.return_value.path = mock_path
+
+        mock_test_recorder_object = MagicMock()
+        mock_test_recorder.return_value = mock_test_recorder_object
+
+        mock_suite_object.result_data.__iter__.return_value = [MagicMock(), MagicMock()]
+
+        runner = IntegTestRunnerOpenSearchDashboards(self.args, self.test_manifest)
+
+        # call the test target
+        results = runner.run()
+
+        self.assertEqual(results["sql-ci-group-1"], mock_test_results)
+        self.assertEqual(results["sql-ci-group-2"], mock_test_results)
+        self.assertEqual(results["sql-ci-group-3"], mock_test_results)
+
+        mock_suite_object.result_data.__iter__.assert_called()
+        mock_test_recorder_object.test_results_logs.generate_component_yml.assert_called()
+
+        expected_call = call(
+            mock_properties_dependency_object.dependency_installer,
+            mock_properties_object.dependency_installer,
+            mock_component,
+            mock_test_config,
+            mock_properties_dependency_object.bundle_manifest,
+            mock_properties_object.bundle_manifest,
+            mock_properties_dependency_object.build_manifest,
+            mock_properties_object.build_manifest,
+            mock_path,
+            mock_test_recorder_object
+        )
+        mock_suite.assert_has_calls([expected_call, expected_call, expected_call], any_order=True)

